tsdownsample/downsampling_interface.py

```diff
@@ -1,57 +1,108 @@
+"""AbstractDownsampler interface-class, subclassed by concrete downsamplers."""
+
+__author__ = "Jeroen Van Der Donckt"
+
+import re
 import warnings
 from abc import ABC, abstractmethod
 from types import ModuleType
-from typing import Callable
+from typing import Callable, List, Optional, Tuple, Union
 
 import numpy as np
-import pandas as pd
 
 
-class DownsampleInterface(ABC):
-    def __init__(self, name: str) -> None:
-        self.name = name
+class AbstractDownsampler(ABC):
```

Comment:
So ideally, other packages which want to integrate `tsdownsample` should use this interface.

Reply:
Exactly!